### PR TITLE
Add Truffleruby head to CI

### PR DESCRIPTION
**Pull request by gogainda**

### CODE REVIEW

**Comment by avdv** [comment]
FTR, the build failed:
> 62 examples, 30 failures
see https://travis-ci.org/github/athityakumar/colorls/jobs/725070795

Also, execution is slow, really slow (as expected):

> `$ colorls` 6.03 s

**Comment by gogainda** [comment]
encoding related issue is fixed https://github.com/oracle/truffleruby/pull/2095

**Comment by avdv** [comment]
This time, the tests ran successfully, but execution of colorls failed when using the `--gs` flag:
```
$ colorls --gs

<internal:core> core/truffle/process_operations.rb:470:in `block (2 levels) in posix_spawnp': undefined method `<' for nil:NilClass (NoMethodError)

	from <internal:core> core/truffle/process_operations.rb:469:in `map'

	from <internal:core> core/truffle/process_operations.rb:469:in `block in posix_spawnp'

	from <internal:core> core/truffle/process_operations.rb:466:in `each_pair'

	from <internal:core> core/truffle/process_operations.rb:466:in `posix_spawnp'

	from <internal:core> core/truffle/process_operations.rb:439:in `spawn'

	from <internal:core> core/truffle/process_operations.rb:110:in `spawn'

	from <internal:core> core/io.rb:853:in `popen'

	from /home/travis/build/athityakumar/colorls/lib/colorls/git.rb:55:in `git_prefix'

	from /home/travis/build/athityakumar/colorls/lib/colorls/git.rb:9:in `status'

	from /home/travis/build/athityakumar/colorls/lib/colorls/core.rb:226:in `process_git_status_details'

	from /home/travis/build/athityakumar/colorls/lib/colorls/core.rb:22:in `initialize'

	from /home/travis/build/athityakumar/colorls/lib/colorls/flags.rb:48:in `new'

	from /home/travis/build/athityakumar/colorls/lib/colorls/flags.rb:48:in `block in process'

	from <internal:core> core/enumerable.rb:339:in `each_with_index'

	from /home/travis/build/athityakumar/colorls/lib/colorls/flags.rb:43:in `process'

	from /home/travis/build/athityakumar/colorls/exe/colorls:26:in `<top (required)>'

	from <internal:core> core/kernel.rb:402:in `load'

	from <internal:core> core/kernel.rb:402:in `load'

	from /home/travis/build/athityakumar/colorls/bin/colorls:29:in `<main>'

The command "colorls --gs" exited with 1.
```

**Comment by gogainda** [comment]
Strange, that it didn't fail in CI, prb it's Mac OS specific

**Comment by avdv** [comment]
> Strange, that it didn't fail in CI, prb it's Mac OS specific

Unfortunately, integration between Travis and GitHub is broken (since a few months already) and I can't fix it.

The job did fail: https://travis-ci.org/github/athityakumar/colorls/jobs/726242476

**Comment by gogainda** [comment]
it could be truffleruby specific issue, will fill out an issue there

**Comment by avdv** [comment]
The last problem seems to be fixed (:tada:), in lieu of a new one:
https://travis-ci.org/github/athityakumar/colorls/builds/727860910#L344
```
undefined method `<<' for #<Truffle::Digest:0x2aaf8>

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/result_cache.rb:183:in `rubocop_checksum'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/result_cache.rb:94:in `initialize'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:135:in `new'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:135:in `cached_result'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:140:in `file_offense_cache'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:127:in `file_offenses'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:118:in `process_file'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:97:in `block in each_inspected_file'

<internal:core> core/enumerable.rb:448:in `inject'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:96:in `each_inspected_file'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:82:in `inspect_files'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:43:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/command.rb:11:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/environment.rb:18:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli.rb:65:in `run_command'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli.rb:72:in `execute_runners'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli.rb:41:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/exe/rubocop:13:in `block in <top (required)>'

/home/travis/.rvm/rubies/truffleruby-head/lib/mri/benchmark.rb:308:in `realtime'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/exe/rubocop:12:in `<top (required)>'

<internal:core> core/kernel.rb:402:in `load'

<internal:core> core/kernel.rb:402:in `load'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/bin/rubocop:23:in `<top (required)>'

<internal:core> core/kernel.rb:402:in `load'

<internal:core> core/kernel.rb:402:in `load'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in `kernel_load'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:28:in `run'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli.rb:476:in `exec'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor.rb:399:in `dispatch'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli.rb:30:in `dispatch'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/base.rb:476:in `start'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli.rb:24:in `start'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/exe/bundle:46:in `block in <top (required)>'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/friendly_errors.rb:123:in `with_friendly_errors'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/exe/bundle:34:in `<top (required)>'

<internal:core> core/kernel.rb:402:in `load'

<internal:core> core/kernel.rb:402:in `load'

/home/travis/.rvm/gems/truffleruby-head/bin/bundle:23:in `<main>'
```
which is probably caused by updating rubocop to version 0.91.0.

**Comment by avdv** [comment]
This time, it build: https://travis-ci.org/github/athityakumar/colorls/builds/729596207 :tada: 

Now what? Nobody would want to use colorls with truffleruby because of its ~sluggishness~ start/runtime overhead.

Did you have anything in mind with this specifically?

**Comment by gogainda** [comment]
Currently we are tracking all gems that support truffleruby, so if it's not a problem I would like to add this gem as well. To do that truffleruby should be part of a pipeline. Regarding performance, I believe it's a current goal for the team and they are currently working on that

**Comment by avdv** [comment]
OK. How about adding it with `fast_finish: true` and `allow_failures` enabled for truffleruby?

**Comment by gogainda** [comment]
@avdv addressed your commet, pls review/merge

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,11 @@ rvm:
   - '2.5'
   - '2.6'
   - '2.7'
+  - 'truffleruby-head'
+matrix:
```
Comment (avdv) [inline .travis.yml:16]: `matrix` is an old name for `jobs`. (There's a validation message: 
    <code>root</code>: key <code>matrix</code> is an alias for <code>jobs</code>, using <code>jobs</code> in the build validation config.)
  

```suggestion

jobs:
```

**Review by avdv — CHANGES_REQUESTED** [review]

**Review by avdv — APPROVED** [review]

**Comment by avdv** [comment]
Thanks @gogainda

**Pull request merged**